lib/Parser/Scan.cpp

```diff
@@ -2087,36 +2092,14 @@ tokens Scanner<EncodingPolicy>::ScanCore(bool identifyKwds)
             case '-':
                 p++;
                 token = tkDec;
-                if (m_fHtmlComments)
+                if (!m_fIsModuleCode)
                 {
                     int i = 0;
                     while ('-' == PeekFirst(p + i, last)) //Have already seen --, skip any further - characters
                         i++;
-                    if ('>' == PeekFirst(p + i++, last)) //This means we've got a --------------------------->.
```

Comment:
Sorry I missed this in my most recent commit. I don't think this is actually allowed in the spec.